app/javascript/components/mentoring/discussion/DiscussionPostList.tsx

```diff
@@ -1,32 +1,55 @@
-import React, { useEffect } from 'react'
+import React, { useEffect, useMemo } from 'react'
 import { useRequestQuery } from '../../../hooks/request-query'
 import { DiscussionPost, DiscussionPostProps } from './DiscussionPost'
 import { DiscussionPostChannel } from '../../../channels/discussionPostChannel'
 import { Loading } from '../../common/Loading'
 
+type Iteration = {
+  idx: number
+  posts: DiscussionPostProps[]
+}
+
 export const DiscussionPostList = ({
   endpoint,
   discussionId,
-  iterationIdx,
 }: {
   endpoint: string
   discussionId: number
-  iterationIdx: number
 }): JSX.Element | null => {
   const { isSuccess, isLoading, data, refetch } = useRequestQuery<{
     posts: DiscussionPostProps[]
   }>(endpoint, { endpoint: endpoint, options: {} })
 
+  const iterations = useMemo(() => {
+    if (!data) {
+      return []
+    }
+
+    return data.posts.reduce<Iteration[]>((iterations, post) => {
+      const iteration = iterations.find(
```

Comment:
I usually use `result` or the plural as first argument, so it doesn't shadow the outer variable name.